src/epoch/initializers.rs

```diff
@@ -0,0 +1,409 @@
+        Self::from_tai_duration(Duration::from_parts(centuries, nanoseconds))
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided TAI seconds since 1900 January 01 at midnight
+    pub fn from_tai_seconds(seconds: f64) -> Self {
+        assert!(
+            seconds.is_finite(),
+            "Attempted to initialize Epoch with non finite number"
+        );
+        Self::from_tai_duration(seconds * Unit::Second)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided TAI days since 1900 January 01 at midnight
+    pub fn from_tai_days(days: f64) -> Self {
+        assert!(
+            days.is_finite(),
+            "Attempted to initialize Epoch with non finite number"
+        );
+        Self::from_tai_duration(days * Unit::Day)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided UTC seconds since 1900 January 01 at midnight
+    pub fn from_utc_duration(duration: Duration) -> Self {
+        Self::from_duration(duration, TimeScale::UTC)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided UTC seconds since 1900 January 01 at midnight
+    pub fn from_utc_seconds(seconds: f64) -> Self {
+        Self::from_utc_duration(seconds * Unit::Second)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided UTC days since 1900 January 01 at midnight
+    pub fn from_utc_days(days: f64) -> Self {
+        Self::from_utc_duration(days * Unit::Day)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided duration since 1980 January 6 at midnight
+    pub fn from_gpst_duration(duration: Duration) -> Self {
+        Self::from_duration(duration, TimeScale::GPST)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided duration since 1980 January 6 at midnight
+    pub fn from_qzsst_duration(duration: Duration) -> Self {
+        Self::from_duration(duration, TimeScale::QZSST)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided duration since August 21st 1999 midnight
+    pub fn from_gst_duration(duration: Duration) -> Self {
+        Self::from_duration(duration, TimeScale::GST)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided duration since January 1st midnight
+    pub fn from_bdt_duration(duration: Duration) -> Self {
+        Self::from_duration(duration, TimeScale::BDT)
+    }
+
+    #[must_use]
+    pub fn from_mjd_tai(days: f64) -> Self {
+        Self::from_mjd_in_time_scale(days, TimeScale::TAI)
+    }
+
+    pub fn from_mjd_in_time_scale(days: f64, time_scale: TimeScale) -> Self {
+        assert!(
+            days.is_finite(),
+            "Attempted to initialize Epoch with non finite number"
+        );
+        Self {
+            duration: (days - MJD_J1900) * Unit::Day,
+            time_scale,
+        }
+    }
+
+    #[must_use]
+    pub fn from_mjd_utc(days: f64) -> Self {
+        Self::from_mjd_in_time_scale(days, TimeScale::UTC)
```

Comment:
This function previously called `from_mjd_tai` and then manually applied the leap seconds. Now it calls the `from_mjd_in_time_scale` function which correctly sets the time scale without changing the duration.